filament/backend/src/vulkan/VulkanDriver.cpp

```diff
@@ -1558,10 +1534,7 @@ void VulkanDriver::readPixels(Handle<HwRenderTarget> src, uint32_t x, uint32_t y
     vkMapMemory(device, stagingMemory, 0, VK_WHOLE_SIZE, 0, (void**) &srcPixels);
     srcPixels += subResourceLayout.offset;
 
-    // TODO: investigate why this Y-flip exists. This conditional seems to work with both
-    // test_ReadPixels.cpp (readpixels from a normal render target with texture attachment) and
-    // viewer_basic_test.cc (readpixels from an offscreen swap chain)
-    const bool flipY = srcTexture ? true : false;
+    const bool flipY = false;
```

Comment:
So there's no longer a difference with surfaces versus textures in regards to flipping y?

Reply:
Yes I tested ReadPixels with headless and non-headless and they both seem right side up now.

To be honest I never understood why this was necessary in the first place.  Maybe there was a bug in older versions of Molten, or something like that.

Reply:
👍